pkg/planner/planner.go

```diff
@@ -333,11 +333,11 @@ func (p *Planner) loadClusterAgent(cluster *rkev1.RKECluster) ([]byte, error) {
 		return nil, err
 	}
 
-	for _, token := range tokens {
-		return DownloadClusterAgentYAML(p.ctx, url, ca, token.Status.Token)
+	if len(tokens) == 0 {
+		return nil, fmt.Errorf("no cluster registration token found")
```

Comment:
The original code looped until it found proper token to return, you are now assuming its always at 0.

Reply:
I don't believe the original code looped at all: it returned on the first token every time.

It seemed to be a less explicit way of doing what I am doing: if there are no tokens the loop was skipped and the error was returned. If there were any tokens, it the loop returned on the first one.

Reply:
Hmm, seem to have same behavior in rancher. Maybe we only ever get one.